apps/api/src/app/step-schemas/step-schemas.controller.ts

```diff
@@ -1,45 +1,48 @@
-import { ClassSerializerInterceptor, Controller, Get, Param, UseInterceptors } from '@nestjs/common';
+import { ClassSerializerInterceptor, Controller, Get, Query, UseInterceptors } from '@nestjs/common';
 
 import { UserSessionData } from '@novu/shared';
 import { ExternalApiAccessible, UserSession } from '@novu/application-generic';
 import { StepType } from '@novu/framework';
 
-import { ApiOperation, ApiParam } from '@nestjs/swagger';
-import { GetStepSchemaCommand } from './usecases/get-step-schema/get-step-schema.command';
+import {
+  GetStepTypeSchemaCommand,
+  GetExistingStepSchemaCommand,
+} from './usecases/get-step-schema/get-step-schema.command';
 import { UserAuthentication } from '../shared/framework/swagger/api.key.security';
 import { GetStepSchema } from './usecases/get-step-schema/get-step-schema.usecase';
-import { SchemaTypeDto } from './dtos/schema-type.dto';
+import { StepSchemaDto } from './dtos/step-schema.dto';
 
 @Controller('/step-schemas')
 @UserAuthentication()
 @UseInterceptors(ClassSerializerInterceptor)
 export class StepSchemasController {
   constructor(private getStepDefaultSchemaUsecase: GetStepSchema) {}
 
-  @Get('/:stepType')
-  @ApiOperation({
-    summary: 'Get step schema',
-    description: 'Get the schema for a step type',
-  })
-  @ApiParam({
-    name: 'stepType',
-    type: String,
-    description: 'The type of step to get the schema for.',
-  })
+  @Get()
   @ExternalApiAccessible()
-  async getStepSchema(
+  async getWorkflowStepSchema(
     @UserSession() user: UserSessionData,
-    @Param('stepType') stepType: StepType
-  ): Promise<SchemaTypeDto> {
-    const schema = await this.getStepDefaultSchemaUsecase.execute(
-      GetStepSchemaCommand.create({
-        organizationId: user.organizationId,
-        environmentId: user.environmentId,
-        userId: user._id,
-        stepType,
-      })
-    );
+    @Query('stepType') stepType: StepType,
+    @Query('workflowId') workflowId: string,
+    @Query('stepId') stepId: string
+  ): Promise<StepSchemaDto> {
+    return await this.getStepDefaultSchemaUsecase.execute(this.createCommand(user, stepType, workflowId, stepId));
+  }
 
-    return { schema };
+  private createCommand(user: UserSessionData, stepType: StepType, workflowId: string, stepId: string) {
```

Comment:
Would it make sense to move this logic into the use case class? Keeping the controller simple and free of business logic could enhance readability and maintainability.

Reply:
Love the idea of the clean controller. i will move it to the command file as it is its responsibility